src/game_battlealgorithm.cpp

```diff
@@ -1292,19 +1292,40 @@ bool Game_BattleAlgorithm::Skill::Execute() {
 				}
 			}
 
-			if (skill.affect_sp) {
+			if (skill.affect_sp && Utils::PercentChance(to_hit)) {
 				this->sp = std::min<int>(effect, GetTarget()->GetSp());
 			}
 
-			if (skill.affect_attack)
+			if (skill.affect_attack && Utils::PercentChance(to_hit))
 				this->attack = std::max<int>(0, std::min<int>(effect, GetTarget()->GetAtk() - (GetTarget()->GetBaseAtk() + 1) / 2));
-			if (skill.affect_defense)
+			if (skill.affect_defense && Utils::PercentChance(to_hit))
 				this->defense = std::max<int>(0, std::min<int>(effect, GetTarget()->GetDef() - (GetTarget()->GetBaseDef() + 1) / 2));
-			if (skill.affect_spirit)
+			if (skill.affect_spirit && Utils::PercentChance(to_hit))
 				this->spirit = std::max<int>(0, std::min<int>(effect, GetTarget()->GetSpi() - (GetTarget()->GetBaseSpi() + 1) / 2));
-			if (skill.affect_agility)
+			if (skill.affect_agility && Utils::PercentChance(to_hit))
 				this->agility = std::max<int>(0, std::min<int>(effect, GetTarget()->GetAgi() - (GetTarget()->GetBaseAgi() + 1) / 2));
 
+			if (skill.affect_hp) {
+				if (skill.affect_sp) {
+					if (GetAffectedHp() == -1 && GetAffectedSp() == -1) {
+						this->success = false;
+						return this->success;
+					}
```

Comment:
Can you explain why this is needed? It's not mentioned in the commit.

Also, this is written badly, this entire nest of conditions can be replaced with a single if statement.
```
if ((skill.affect_hp && GetAffectedHp() == -1) || (skill.affect_sp && GetAffectedSp() == -1))
```

Reply:
The reason for this is explained in the fifth bullet point of the description of this PR. Special accuracy checks are made if a skill targets enemies. If affect_hp without affect_sp is set, then the whole skill fails if the accuracy check for HP fails. If affect_sp without affect_hp is set, then the whole skill fails if the accuracy check for SP fails. If affect_hp and affect_sp are set, then the whole skill fails if both accuracy checks fail. If neither affect_hp nor affect_sp are set then the accuracy checks are performed normally.

Reply:
I have tried out your proposal but unfortunately it doesn't work if both affect_hp and affect_sp are set. This makes the skill fail if one of both accuracy checks fail. But both checks need to fail to make the whole skill fail if both affect_hp and affect_sp are set.